camera/camera_manager.cc

```diff
@@ -0,0 +1,85 @@
+//%deps(yaml-cpp)
+
+#include "camera/camera_manager.hh"
+
+namespace jet {
+
+CameraManager::CameraManager() {
+  load_configs();
+  if (camera_map_.size() == 0) {
+    std::string err = std::string("No camera configs read!");
+    throw std::runtime_error(err);
+  }
+}
+
+void CameraManager::load_configs() {
+  // directory iteration from http://www.martinbroadhurst.com/list-the-files-in-a-directory-in-c.html
+  DIR* dir_path = opendir(config_dir.c_str());
+  if (dir_path == NULL) {
+    std::string err = std::string("Unable to open ") + config_dir;
+    throw std::runtime_error(err);
+  }
+  const struct dirent* dp;
+  while ((dp = readdir(dir_path)) != NULL) {
+    std::string filename = dp->d_name;
+    if (filename == "." || filename == ".."){ 
+      continue;
+    }
+    std::optional<YAML::Node> config = read_YAML(config_dir + filename);
+    std::optional<Camera> camera = parse_config(*config);
+    if (camera) {
+      camera_map_[camera->serial_number] = *camera;
+    }
+  }
+}
+
+std::optional<YAML::Node> CameraManager::read_YAML(std::string filepath) {
+  try {
+    return YAML::LoadFile(filepath);
+  } catch (YAML::BadFile e) {
+    std::string err = std::string("Could not find YAML file ") + filepath;
+    throw std::runtime_error(err);
+  }
+}
+
+std::optional<Camera> CameraManager::parse_config(YAML::Node cfg) {
+  Camera camera;
+  // path & video index
+  camera.serial_number = cfg["serial_number"].as<std::string>();
+  camera.v4l_path = "/dev/v4l/by-id/" + cfg["v4l_path"].as<std::string>();
+  camera.video_index = follow_v4l_path(camera.v4l_path);
+  if (camera.video_index == -1) {
+    return std::nullopt;
+  }
+  // calibration values
+  std::vector<double> camera_matrix_values = cfg["camera_matrix"].as<std::vector<double>>();
```

Comment:
`const`

Reply:
unfortunately the cv::Mat constructor doesn't like the const